modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java

```diff
@@ -107,11 +118,38 @@ public IgniteRepositoryFactory(String springCfgPath) {
 
         Assert.hasText(annotation.cacheName(), "Set a name of an Apache Ignite cache using @RepositoryConfig " +
             "annotation to map this repository to the underlying cache.");
+        String cacheName = null;
+        if (!looksLikeExpression(annotation.cacheName())) {
+            cacheName = annotation.cacheName();
+        }
+        else {
+            SpelExpressionParser parser = new SpelExpressionParser();
+            Expression expression = parser.parseExpression(annotation.cacheName());
+            StandardEvaluationContext ec = new StandardEvaluationContext();
+            ec.setBeanResolver(new BeanFactoryResolver(ctx.getAutowireCapableBeanFactory()));
+            cacheName = (String)expression.getValue(ec);
+        }
+        repoToCache.put(repoItf, cacheName);
 
         repoToCache.put(repoItf, annotation.cacheName());
 
         return super.getRepositoryMetadata(repoItf);
     }
+    /**
+     * The method tryes to identify SpEL extression
+     * @see <a href="https://docs.spring.io/spring/docs/5.0.16.RELEASE/spring-framework-reference/core.html#expressions">SpEL</a>
+     * @param expression
+     * @return
+     */
+
+    private boolean looksLikeExpression(String expression) {
```

Comment:
Maybe rename  to: isSpelExpression ?

Reply:
The method checks the expression for attributes of SpEL. You are right. Fixed.

Reply:
fixed